packages/core/content-manager/server/services/utils/configuration/layouts.js

```diff
@@ -7,9 +7,16 @@ const { isListable, hasEditableAttribute, hasRelationAttribute } = require('./at
 const DEFAULT_LIST_LENGTH = 4;
 const MAX_ROW_SIZE = 12;
 
-const isAllowedFieldSize = (type, size) => {
+const isAllowedFieldSize = (type, size, customFieldUid = null) => {
   const { getFieldSize } = getService('field-sizes');
-  const fieldSize = getFieldSize(type);
+  let fieldSize = getFieldSize(type);
+
+  if (customFieldUid) {
+    const customField = strapi.container.get('custom-fields').get(customFieldUid);
+    if (customField.inputSize) {
+      fieldSize = customField.inputSize;
+    }
+  }
```

Comment:
I am not a backend developer, but wouldn't it be cleaner if `getFieldSize` returns the proper field-size regardless of the type? Proposing to move this bit into the function itself.

It would also be great if you could add a test for it 🙏🏼